### PR TITLE
Update local configs to fix issues with incorrect addresses

### LINKED ISSUE #5206: Ring instance addresses are all the same running 3 single-binaries in docker
**Describe the bug**
When running 3 single binaries in a docker compose setup, using memberlist for the ingester ring and replication factor of 3, all 3 instances report an address of `127.0.0.1:9095`. As a result, when iterating for the replica set in the distributor, 3 requests are sent to localhost rather than being distributed across other nodes as expected.

**To Reproduce**
1. Start Loki in docker compose with the following config
```yaml
  loki:
    image: grafana/loki:2.4.2
    command:
      - -memberlist.join=loki:7946
      - -ring.store=memberlist
      - -distributor.replication-factor=3
      - -config.file=/etc/loki/local-config.yaml
    ports:
      - 3100
      - 9095
    deploy: &replicated
      mode: replicated
      replicas: 3
    networks: &loki-dns
      default:
        aliases:
          - loki
```

2. Go to the `/ring` endpoint
3. See 3 instances all with address `127.0.0.1:9095`

**Expected behavior**
I would expect each instance in the ring to have a unique IP address that is accessible from all the other instances in the ring.

**Environment:**
 - Docker compose

**Screenshots, Promtail config, or terminal output**
Screenshot from 2022-01-21 11-29-50

### PR DESCRIPTION
**Pull request by trevorwhitney**
**What this PR does / why we need it**:

Remove the `instance_addr` property from the local and docker configs. Hard-coding this to `127.0.0.1` breaks any attempt to scale, as all instances in the ring will report the wrong address.

**Which issue(s) this PR fixes**:
Fixes #5206 

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated
- [ ] Add an entry in the `CHANGELOG.md` about the changes.

### CODE REVIEW

**Comment by trevorwhitney** [comment]
There may be a larger discussion here, but specifying an `instance_addr` of `127.0.0.1` will only ever work if there is a single node. This may be reasonable of the `loki-local-config.yaml`, but I don't think it makes sense for the docker one. The reason specifying an `instance_addr` is useful is in case loki is running on a non-standard env, and so it has trouble finding an IP based on the `interface_names`. Again, in docker, this is less likely to occur.

File: cmd/loki/loki-local-config.yaml [hunk 1]
```diff
@@ -12,7 +12,6 @@ common:
       rules_directory: /tmp/loki/rules
   replication_factor: 1
   ring:
-    instance_addr: 127.0.0.1
```
Comment (kavirajk) [inline cmd/loki/loki-local-config.yaml:15]: Make sense to remove it from docker config. But i don't understand why from local-config as well? (we use this only to run in single node AFAIK).

File: cmd/loki/loki-docker-config.yaml [hunk 2]
```diff
@@ -11,7 +11,6 @@ common:
       rules_directory: /loki/rules
   replication_factor: 1
   ring:
-    instance_addr: 127.0.0.1
```
Comment (kavirajk) [inline cmd/loki/loki-docker-config.yaml:14]: Q: Curious, what happens if we don't provide this config? Do Loki still choose `127.0.0.1` with random port?

Comment (trevorwhitney) [reply cmd/loki/loki-docker-config.yaml:14]: Loki will pick an address from the interface list, which from my experience in Docker is usually not 127.0.0.1, but some real IP on the docker network being used. Maybe this is on the case for compose or swarm?

File: cmd/loki/loki-local-config.yaml (hunk 1 above)
Comment (trevorwhitney) [reply cmd/loki/loki-local-config.yaml:15]: I agree, I'm going to remove this change.

File: cmd/loki/loki-docker-config.yaml (hunk 2 above)
Comment (kavirajk) [reply cmd/loki/loki-docker-config.yaml:14]: I meant in docker. Make sense :+1:

**Review by kavirajk — APPROVED** [review]
LGTM

**Pull request merged**